src/Impostor.Server/Net/Client.cs

```diff
@@ -35,14 +35,72 @@ public Client(ILogger<Client> logger, IOptions<AntiCheatConfig> antiCheatOptions
             _customMessageManager = customMessageManager;
         }
 
-        public override async ValueTask<bool> ReportCheatAsync(CheatContext context, string message)
+        public override async ValueTask<bool> ReportCheatAsync(CheatContext context, CheatCategory category, string message)
         {
             if (!_antiCheatConfig.Enabled)
             {
                 return false;
             }
 
-            _logger.LogWarning("Client {Name} ({Id}) was caught cheating: [{Context}] {Message}", Name, Id, context.Name, message);
+            if (_antiCheatConfig.ForbidHostCheating && Player != null && Player.IsHost)
+            {
+                return false;
+            }
+
+            switch (category)
+            {
+                case CheatCategory.ProtocolExtension:
+                    if (!_antiCheatConfig.ForbidProtocolExtensions)
+                    {
+                        return false;
+                    }
+
+                    break;
+                case CheatCategory.GameFlow:
+                    if (!_antiCheatConfig.EnableGameFlowChecks)
+                    {
+                        return false;
+                    }
+
+                    break;
+                case CheatCategory.MustBeHost:
+                    if (!_antiCheatConfig.EnableMustBeHostChecks)
+                    {
+                        return false;
+                    }
+
+                    break;
+                case CheatCategory.Limit:
+                    if (!_antiCheatConfig.EnableLimitChecks)
+                    {
+                        return false;
+                    }
+
+                    break;
+                case CheatCategory.Ownership:
+                    if (!_antiCheatConfig.EnableOwnershipChecks)
+                    {
+                        return false;
+                    }
+
+                    break;
+                case CheatCategory.Role:
+                    if (!_antiCheatConfig.EnableRoleChecks)
+                    {
+                        return false;
+                    }
+
+                    break;
+                case CheatCategory.Target:
+                    if (!_antiCheatConfig.EnableTargetChecks)
+                    {
+                        return false;
+                    }
+
+                    break;
+            }
```

Comment:
```suggestion
            var isCategoryEnabled = category switch
            {
                CheatCategory.ProtocolExtension => _antiCheatConfig.ForbidProtocolExtensions,
                CheatCategory.GameFlow => _antiCheatConfig.EnableGameFlowChecks,
                CheatCategory.MustBeHost => _antiCheatConfig.EnableMustBeHostChecks,
                CheatCategory.Limit => _antiCheatConfig.EnableLimitChecks,
                CheatCategory.Ownership => _antiCheatConfig.EnableOwnershipChecks,
                CheatCategory.Role => _antiCheatConfig.EnableRoleChecks,
                CheatCategory.Target => _antiCheatConfig.EnableTargetChecks,
                CheatCategory.Other => true,
                _ => throw new ArgumentOutOfRangeException(nameof(category), category, null),
            };

            if (!isCategoryEnabled)
            {
                return false;
            }
```

Reply:
Accepted but with a different discard pattern, I don't think throwing exceptions from the API is acceptable. It may be desireable to make changes in the CheatCategory enum in future, and throwing on unknown categories could cause old plugins to break

Reply:
>  could cause old plugins to break

In what way?

Reply:
If we ever remove an old category, but that's an API break. Considering the average plugin quality and because C# allows casting numbers to enums crashing on an invalid anticheat category seems harsh.